omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -284,6 +290,19 @@ public String handleSubmission(WebRequest request, HttpSession httpSession, Mode
 				us.changeQuestionAnswer(user, secretQuestion, secretAnswer);
 			}
 			
+			//Check if admin wants the person associated with the user to be added to the Provider Table
+			if(addToProviderTableOption != null) {
+				Set<Role> userRoles = user.getRoles();
+				for (Role role : userRoles) {
+					if (role.getRole().equals("Provider")) { //If 'Provider' role is selected, we add the User to the Provider Table
+						Provider provider = new Provider();
+						provider.setPerson(user.getPerson());
+						provider.setName(user.getName());
+						provider.setIdentifier(user.getSystemId());
+						Context.getProviderService().saveProvider(provider);
```

Comment:
Is this happening after or before you save the user record? Because it should be after